server/src/data/channels-data.js

```diff
@@ -22,14 +22,18 @@ class ChannelsData extends DataBase {
     let channel = await models.Channel.findOne({
       where: { name: input.name },
     });
-
-    if (!channel) {
-      channel = await models.Channel.create({
-        ...input,
-        creatorId: user.agentId,
-      });
-      await this.agentsChannelsData.createOne(channel.id, user.agentId);
+    if (channel) {
+      return Promise.reject(new Error('Channel exists.'));
     }
+    channel = await models.Channel.create({
+      ...input,
+      creatorId: user.agentId,
+    });
+    await this.agentsChannelsData.createOne(
+      channel.id,
+      user.agentId,
+      models.AgentsChannels.ROLE.ADMIN,
```

Comment:
Can this be done in an ``afterCreate`` callback in the Channel model layer? That seems like where I would more expect it.

Reply:
No, according to our strategy models do not know logic of application work. They only represents data from DB.